cmd/goProbe/config/monitor.go

```diff
@@ -0,0 +1,129 @@
+package config
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/els0r/goProbe/pkg/logging"
+)
+
+const defaultReloadInterval = 5 * time.Minute
+
+// Monitor denotes a config monitor / manager
+type Monitor struct {
+	path   string
+	config *Config
+
+	reloadInterval time.Duration
+
+	sync.RWMutex
+}
+
+// CallbackFn denotes a function to be called upon successful reload of the configuration
+type CallbackFn func(context.Context, Ifaces) ([]string, []string, []string, error)
```

Comment:
Named return parameters could make clearer why the structure of the type is the way it is: to return `added`, `updated` and `deleted`.

Reply:
Yeah, had that initially, just removed it to make the definition shorter (because we don't actually care / use these parameters for the callback in the monitor, they just happen to be there :stuck_out_tongue:) - But I'm game either way and add them.